ext/standard/tests/http/gh9316.phpt

```diff
@@ -0,0 +1,38 @@
+--TEST--
+Bug GH-9316 ($http_response_header is wrong for long status line)
+--SKIPIF--
+<?php require 'server.inc'; http_server_skipif(); ?>
+--INI--
+allow_url_fopen=1
+--FILE--
+<?php
+require 'server.inc';
+
+$responses = array(
+    "data://text/plain,HTTP/1.1 200 Some very long reason-phrase to test that this is properly handled by our code without adding a new header like  Bad: Header\r\nGood: Header\r\n\r\nBody",
+    "data://text/plain,HTTP/1.1 200\r\nGood: Header\r\n\r\nBody",
```

Comment:
```suggestion
    "data://text/plain,HTTP/1.1 200 \r\nGood: Header\r\n\r\nBody",
```
While the *reason* may be absent, the space after the status code must be present:

> The first line of a response message is the status-line, consisting of the protocol version, a space (SP), the status code, and another space and ending with an OPTIONAL textual phrase describing the status code.

>  [status-line](https://httpwg.org/specs/rfc9112.html#status.line) = [HTTP-version](https://httpwg.org/specs/rfc9112.html#http.version) [SP](https://httpwg.org/specs/rfc9112.html#core.rules) [status-code](https://httpwg.org/specs/rfc9112.html#status.line) [SP](https://httpwg.org/specs/rfc9112.html#core.rules) [ [reason-phrase](https://httpwg.org/specs/rfc9112.html#status.line) ]

https://httpwg.org/specs/rfc9112.html#rfc.section.4

Reply:
Yes, I'm aware of that, but I also wanted to test some non conforming status line, and I think we should still accept that (Postel's law). I'll add another "data set".

Reply:
I very strongly disagree with *testing* known-broken behavior. Not actually verifying that the required space is there might be reasonable. But a test implies that accepting the broken response is actually a feature which makes it harder to remove, if it turns out that being lax here causes issues.

See also: https://datatracker.ietf.org/doc/html/draft-thomson-postel-was-wrong-00 (I know it's just a draft, but I agree with the arguments put forward there).

Reply:
Interesting! I wasn't aware of that RFC draft, and while I do not necessarily agree with its reasoning, I'm okay with not testing the non-conforming behavior.